internal/daemon/controller/handlers/targets/credentials.go

```diff
@@ -218,6 +233,27 @@ func staticToSessionCredential(ctx context.Context, cred credential.Static) (*pb
 			return nil, errors.Wrap(ctx, err, op, errors.WithMsg("creating proto struct for ssh private key credential"))
 		}
 
+	case *credstatic.KvCredential:
+		var err error
+		credType = string(credential.KvType)
+		s := map[string]interface{}{}
+		err = json.Unmarshal(c.GetSecrets(), &s)
+		if err != nil {
+			return nil, errors.New(ctx, errors.InvalidParameter, op, "unmarshalling kv secret")
+		}
+		kvSecret, err := structpb.NewStruct(s)
+		credData, err = handlers.ProtoToStruct(
+			&pb.KVCredential{
+				Secret: kvSecret,
+			},
+		)
+		secret = map[string]interface{}{
+			"secret": s,
```

Comment:
Shouldn't `secret` be assigned to `s` here? For the other types, the credential values are top-level, but here they're being put a level down.

Reply:
Sorry, I don't understand the intentions for flattening the map. I believe that the fields provided by the user in the variable s will never be accessed within the code. Kv secrets are only intended for passing along json information, never to validate its' content for a specific structure. I also think that the variable s would never be written to by boundary in the future, so flattening the map would not help.

Reply:
I do think it is reasonable to remove the nested object when displaying the kv secret during a session.

Reply:
I removed the nested secrets field and assigned `s` to `secret`